### PR TITLE
KEYCLOAK-14915 Upgrade registration screen to PF4

### PR DESCRIPTION
**Pull request by mabartos**
JIRA: [KEYCLOAK-14915](https://issues.redhat.com/browse/KEYCLOAK-14915)

Old one:
registerOld

New one:
registerNew

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrorPage.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import static org.keycloak.testsuite.util.UIUtils.getTextFromElement;
+
+/**
+ * @author <a href="mailto:mabartos@redhat.com">Martin Bartos</a>
+ */
+public class RegisterInputErrorPage {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrorPage.java:29]: This is not separate page if I understand correctly? Will it be more appropriate to have name like "RegisterInputErrors" or something like that?

**Review by mposolda — CHANGES_REQUESTED** [review]
Adding one minor comment. Also I am adding @Pepo48 and @vmuzikar as reviewers of this. And it will be good to run UI tests in pipeline with various browsers to doublecheck there is no regression.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrorPage.java (hunk 1 above)
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrorPage.java:29]: @mposolda Yes, you're right. Thanks!

**Comment by mabartos** [comment]
There are some failures in UI tests, but it's not caused by this PR, so these tests were executed successfully for this PR.

**Comment by Pepo48** [comment]
@mabartos first of all - nice job!

The only difference in IE that I see is the missing exclamation mark icon at the right side of the textboxes:
obrázok
But I assume you are already aware of it, don't you? If so, I'm completely fine with it.
Anyway, you may see on the screenshot that the "background overflow" still persists and it is the most visible just on the registration form (obviously because it's the largest form). I think that this is caused by the resolution of the background picture - we could take a look on it as a follow up task. 

None of these I perceive as a blocker thus I'm approving it. Thanks!

**Review by Pepo48 — DISMISSED** [review]

**Review by vmuzikar — COMMENTED** [review]
Found one nitpick. Other than that LGTM. Thank you @mabartos!

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrors.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import static org.keycloak.testsuite.util.UIUtils.getTextFromElement;
+
+/**
+ * @author <a href="mailto:mabartos@redhat.com">Martin Bartos</a>
+ */
+public class RegisterInputErrors {
```
[the remark this answers is not in the document]
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrors.java:29]: @vmuzikar Good point! I'll try to remade that.

[the remark this answers is not in the document]
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrors.java]: I renamed it to `GeneralInputErrors`. I think it's better.

[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrors.java]: IMHO `GeneralInputErrors` is even more confusing. :) It's too ... general? It implies any input field error belongs here, which is not true, right?

I still feel it should be split and placed directly into the POs they belong to (`AccountFields` and `PasswordFields`). But if we leave it in a single separate class, maybe we could call it `ProfileInputErrors` or something? WDYT?

[the remark this answers is not in the document]
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/RegisterInputErrors.java]: @vmuzikar I removed the separate class and divided the methods into `AccountFields` and `PasswordFields`. Now, I'm waiting for tests result.

**Comment by mabartos** [comment]
@vmuzikar Base tests passed. See https://keycloak-jenkins.com/job/universal-test-pipeline-server/2291/

**Review by vmuzikar — DISMISSED** [review]
LGTM, thank you @mabartos!

**Comment by mabartos** [comment]
@vmuzikar @Pepo48 Thanks for approving. I've just squashed the commits.

**Comment by mposolda** [comment]
@mabartos Approving. Will wait for GH actions to finish.

**Comment by mposolda** [comment]
@mabartos Looks like lots of test failures and looks like they are regression of this PR. It seems this will require some additional work?

**Comment by vmuzikar** [comment]
@mposolda I think the GH Actions errors are unrelated to this. I saw the exact same failures in my PR: https://github.com/keycloak/keycloak/pull/7586

**Comment by vmuzikar** [comment]
Although it is quite suspicious that all the errors seem to be related to the error notifications which were refactored in this and other PF4 related PRs.

**Comment by mposolda** [comment]
@vmuzikar Yes, I figured that too and pinged @mabartos about this. It seems to me the GH failures were introduced by 2 previous PRs, which individually were passing, but together there are test failures..

**Comment by mabartos** [comment]
It seems, there is some issue with cache in GH actions.

**Comment by mabartos** [comment]
For sure, I've started pipeline with this PR and all tests passed. I don't know, why the KC CI is failing. It seems that the `themes` module is not still built properly. 

https://keycloak-jenkins.com/job/universal-test-pipeline-server/2301/

**Review by mposolda — APPROVED** [review]

**Pull request merged**